### PR TITLE
PACA-513: Use RIF Column enums to get column names.

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-513](https://jira.cms.gov/browse/PACA-513)

**User Story or Bug Summary:**

To future-proof any typos in the Bridge code when referencing RIF column names, we should reference the column enums already available to us in the bfd-model package rather than redefining them in the Bridge app.

---

### What Does This PR Do?

Replaces literal strings for column names with references to the corresponding enum variable's `.name()` method.  Fields with numbered names are converted from string concatenation to looking up names in a list.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/Mcs.java [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+  public static final String NCH_CARR_CLM_SBMTD_CHRG_AMT =
+      CarrierClaimColumn.NCH_CARR_CLM_SBMTD_CHRG_AMT.name();
+  public static final String ORG_NPI_NUM = CarrierClaimColumn.ORG_NPI_NUM.name();
+
+  /**
+   * The maximum number of diagnosis column names available for {@link ICD_DGNS_CD} and {@link
+   * ICD_DGNS_VRSN_CD}. Used when looping over both sets of fields.
+   */
+  public static final int MAX_DIAGNOSIS_CODES =
+      Math.min(ICD_DGNS_CD.size(), ICD_DGNS_VRSN_CD.size());
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/Mcs.java:41]: In FISS we used `Math.max` for both, should those use `min` as well?

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java [hunk 2]
```diff
@@ -217,18 +214,18 @@ String convertDcn(Parser.Data<String> data) {
    */
   @VisibleForTesting
   void addDiagCodes(FissClaim.Builder claimBuilder, Parser.Data<String> data) {
-    for (int i = 1; i <= MAX_DIAG_CODES; ++i) {
-      final int INDEX = i;
+    for (int i = 1; i <= Fiss.MAX_DIAG_CODES; ++i) {
+      final int INDEX = i - 1;
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java:218]: I think `i` is not being used anymore basically, should this just be changed to `int i = 0; i < Fiss.MAX_DIAG_CODES` for simplicity? (Same for the other similar code snippets in Fiss/McsTransformer)

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java]: Good idea.  I've made the change.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/Mcs.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/Mcs.java:41]: Thanks for catching that.  Yes those should have used `min` as well.  It's fixed now.

**Review by KyRobbins — DISMISSED** [review]

**Review by ericbuckley — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/ModelUtil.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+/** Holds common utility methods for model classes. */
+class ModelUtil {
+  /**
+   * We need a list of the names of all enum values that end with a number and have the same base
+   * name as the provided example. Names fit the pattern ICD_PRCDR_CD1, ICD_PRCDR_CD2, etc.
+   *
+   * @param enumValues expected to be the result of a call to {@link Enum#values}
+   * @param baseValue a template enum value to extract the base name from
+   * @return unmodifiable {@link List} of names of all numbered enums with same base name
+   * @param <E> a RIF column enum class containing our enum values
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/model/ModelUtil.java:16]: Nitpick, return should come last

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/FissTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package gov.cms.bfd.pipeline.bridge.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import gov.cms.bfd.model.rif.InpatientClaimColumn;
+import org.junit.jupiter.api.Test;
+
+public class FissTest {
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/FissTest.java:8]: Missing class level doc

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/McsTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package gov.cms.bfd.pipeline.bridge.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import gov.cms.bfd.model.rif.CarrierClaimColumn;
+import org.junit.jupiter.api.Test;
+
+public class McsTest {
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/McsTest.java:8]: Class level doc

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/ModelUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package gov.cms.bfd.pipeline.bridge.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import gov.cms.bfd.model.rif.InpatientClaimColumn;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class ModelUtilTest {
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/ModelUtilTest.java:9]: Class doc here too

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/ModelUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  /**
+   * Lists of names produced by {@link ModelUtil#listNumberedEnumNames} are expected to all have the
+   * exact same name followed by an integer from 1 to the number of items in this list. This helper
+   * method verifies that this pattern holds for the provided list of names and first enum name.
+   *
+   * @param names names produced by {@link ModelUtil#listNumberedEnumNames}
+   * @param firstNameInSequence first enum value whose name appears in the list
+   */
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/test/java/gov/cms/bfd/pipeline/bridge/model/ModelUtilTest.java:29]: Thanks for the great documentation here and on the other methods in this PR! Makes it very east to follow what's going on.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java [hunk 8]
```diff
@@ -193,18 +191,19 @@ String convertIcn(Parser.Data<String> data) {
    */
   @VisibleForTesting
   void addDiagnosisCodes(McsClaim.Builder claimBuilder, Parser.Data<String> data, String icn) {
-    for (int i = 1; i <= MAX_DIAGNOSIS_CODES; ++i) {
+    for (int i = 0; i < Mcs.MAX_DIAGNOSIS_CODES; ++i) {
+      // We can't use the loop index directly because value must be final in the lambda expression
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java:195]: Very helpful context comment, I was wondering about this as I was looking at it.

**Review by lsmitchell — COMMENTED** [review]
Very much liking the documentation and testing in this PR! A couple minor fixes, but overall looks good.

**Review by christrotteradhoc — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Review by KyRobbins — APPROVED** [review]

**Pull request merged**